Integrations/integration-Active_Directory_Query.yml

```
@@ -0,0 +1,1320 @@
+
+        if args.get('attributes'):
+            custome_attributes = args['attributes'].split(",")
+
+        attributes = set(custome_attributes + default_attributes)
+
+        # neasted search
+        query = "(&(objectCategory={})(objectClass=user)(memberOf:1.2.840.113556.1.4.1941:={}))".format(member_type, group_dn)
+
+        entries = search_with_paging(
+            query,
+            DEFAULT_BASE_DN,
+            attributes=attributes,
+            page_size=DEFAULT_PAGE_SIZE
+        )
+
+        members = [ {'dn': entry['dn'], 'category': member_type} for entry in entries['flat']]
+
+        demisto_entry = {
+            'ContentsFormat': formats['json'],
+            'Type': entryTypes['note'],
+            'Contents': entries['raw'],
+            'ReadableContentsFormat': formats['markdown'],
+            'HumanReadable': tableToMarkdown("Active Directory - Get Group Members", entries['flat']),
+            'EntryContext': {
+                'ActiveDirectory.Groups(obj.dn ==' + group_dn + ')': {
+                    'dn': group_dn,
+                    'members': members
+                }
+            }
+        }
+
+        if member_type == 'person':
+            demisto_entry['EntryContext']['ActiveDirectory.Users(obj.dn == val.dn)'] = entries['flat']
+            demisto_entry['EntryContext']['Account'] = [account_entry(entry, custome_attributes) for entry in entries['flat']]
+        else:
+            demisto_entry['EntryContext']['ActiveDirectory.Computers(obj.dn == val.dn)'] = entries['flat']
+            demisto_entry['EntryContext']['Endpoint'] = [endpoint_entry(entry, custome_attributes) for entry in entries['flat']]
+
+        demisto.results(demisto_entry)
+
+    ''' DATABASE OPERATIONS '''
+
+    ''' CREATE OBJECT'''
+
+    def create_user():
+        args = demisto.args()
+
+        object_classes = ["top", "person", "organizationalPerson", "user"]
+        user_dn = args.get('user-dn')
+        username = args.get("username")
+        password = args.get("password")
+        custome_attributes = args.get('custom-attributes')
+        attributes = {
+            "samAccountName": username
+        }
+
+        # set common user attributes
+        if args.get('display-name'):
+            attributes['displayName'] = args['display-name']
+        if args.get('description'):
+            attributes['description'] = args['description']
+        if args.get('email'):
+            attributes['mail'] = args['email']
+        if args.get('telephone-number'):
+            attributes['telephoneNumber'] = args['telephone-number']
+        if args.get('title'):
+            attributes['title'] = args['title']
+
+        # set user custome attributes
+        if custome_attributes:
+            try:
+                custome_attributes = json.loads(custome_attributes)
+            except Exception as e:
+                demisto.info(e.message)
```

```suggestion
                demisto.info(str(e))
```
`message` of exception has been removed in python 3.0. See: https://www.python.org/dev/peps/pep-0352/#retracted-ideas